src/repo/iRepository.ts

```diff
@@ -0,0 +1,121 @@
+import { v4 as uuid } from 'uuid';
+const Sequelize = require('sequelize')
+
+const dbConnection = new Sequelize('mysql', 'root', 'supersecretpass', {
+    host: 'localhost',
+    port: 3306,
+    dialect: 'mysql'
+} )
+
+dbConnection.authenticate().then(() => {
+    console.log("Database connected")
+}).catch( () => {
+    console.log("Database couldnt connect")
+})
+
+//define Task data values and types
+export interface Task {
+    id: uuid;
+    createdAt: Date;
+    title: string;
+    dueDate: string;
+    completed: boolean;
+}
+ 
+//generic repository interface
+interface IRepo<Task>{
+    create(T: Task)
+    getAll()
+    getId(id: uuid)
+    getCompleted(val: string)
+    sortBy(val: string)
+    updateById(id: string, updated: Task)
+    removeById(id: string)
+}
+
+
+//implementation of generic repository interface 
+class taskRepository implements IRepo<Task>  {
+    
+    async create(T: Task){
+        //Used ? for values as couldnt directly assign task values and then replaced them beneath
+        const [results, metadata] = await dbConnection.query(
+            'INSERT INTO tasks VALUES ( ?, ?, ?, ?, ?)', {
+                replacements: [T.id, T.createdAt, T.title, T.dueDate, T.completed]
+            } )
```

Comment:
Replacements, A+.

My question now is, WHY do you use replacements? Does the syntax just look cooler?

Reply:
Replacements sanitize the query before using it to make sure it hasn't been modified. This prevents sql injections from occurring by removing any special characters that could have been added to the original query.

Reply:
How do you sanitize inputs?

Reply:
By validating the user input is the correct data type. This could be done using the Joi plugin